src/lib/import-export/export/exporters/default-exporter.ts

```diff
@@ -0,0 +1,102 @@
+import fs from 'fs';
+import fsPromises from 'fs/promises';
+import os from 'os';
+import path from 'path';
+import archiver from 'archiver';
+import { ExportOptions, BackupContents, Exporter } from '../types';
+
+export class DefaultExporter implements Exporter {
+	private archive!: archiver.Archiver;
+
+	constructor( protected backup: BackupContents ) {}
+
+	async export( options: ExportOptions ): Promise< void > {
+		const output = fs.createWriteStream( options.backupFile );
+		this.archive = this.createArchive( output, options );
+
+		try {
+			await this.addWpConfig();
+			await this.addWpContent( options );
+			await this.addDatabase( options );
+			await this.finalizeArchive();
+		} catch ( error ) {
+			this.archive.abort();
+			throw error;
+		} finally {
+			if ( options.includes.database ) {
+				await this.cleanupTempFiles();
+			}
+		}
+	}
+
+	private createArchive( output: fs.WriteStream, options: ExportOptions ): archiver.Archiver {
+		const isZip = options.backupFile.endsWith( '.zip' );
+		const archive = archiver( isZip ? 'zip' : 'tar', {
+			gzip: ! isZip,
+			gzipOptions: isZip ? undefined : { level: 9 },
+		} );
+
+		output.on( 'close', () => {
+			console.log( `Backup created at: ${ output.path }` );
+		} );
+
+		archive.on( 'warning', ( err ) => {
+			if ( err.code === 'ENOENT' ) {
+				console.warn( 'Archiver warning:', err );
+			} else {
+				throw err;
+			}
+		} );
+
+		archive.on( 'error', ( err ) => {
+			throw err;
+		} );
+
+		archive.pipe( output );
+
+		return archive;
+	}
+
+	private async addWpConfig(): Promise< void > {
+		if ( this.backup.wpConfigFile ) {
+			this.archive = this.archive.file( this.backup.wpConfigFile || '', {
```

Comment:
Since we already check that `this.backup.wpConfigFile` is defined, we won't need to fall back to an empty string.

```suggestion
			this.archive = this.archive.file( this.backup.wpConfigFile, {
```